### PR TITLE
feat: support separate debug & logpath for rules

### PR DESCRIPTION
**Pull request by xjasonlyu**

### CODE REVIEW

**Comment by xjasonlyu** [comment]
@ngjaying This feature has been tested locally and is working as expected.

File: internal/topo/topo.go [hunk 1]
```diff
@@ -120,8 +125,37 @@ func (s *Topo) addEdge(from api.TopNode, to api.TopNode, toType string) {
 // stream starts execution.
 func (s *Topo) prepareContext() {
 	if s.ctx == nil || s.ctx.Err() != nil {
-		contextLogger := conf.Log.WithField("rule", s.name)
-		ctx := kctx.WithValue(kctx.Background(), kctx.LoggerKey, contextLogger)
+		ctxLogger := &logrus.Logger{
```
Comment (ngjaying) [inline internal/topo/topo.go:128]: Are there cases to reuse logger as before?

Comment (xjasonlyu) [reply internal/topo/topo.go:128]: If we reused logger, any changes for rule logger will affect the global logger, which I believe is not what we expect.

Comment (ngjaying) [reply internal/topo/topo.go:128]: That is the default behavior previously so it should be harmless. With this change, will it have problems when writing to the same file simultaneously by multiple rules?

Comment (xjasonlyu) [reply internal/topo/topo.go]: Fixed.

**Review by ngjaying — APPROVED** [review]

**Comment by ngjaying** [comment]
@Yisaer Please check if this breaks the log gc

**Review by Yisaer — CHANGES_REQUESTED** [review]
we clear outdated log when restart eKuiper in `gcOutdatedLog` with logPath defined in config.

When we supported rule-level logPath, I think we should also clear the outdated log.

**Pull request merged**